### ISSUE TITLE
Allow easy recreation.

### ISSUE THREAD

**Issue by ionelmc** [cookiecutter/cookiecutter#235]
Cookiecutter should dump the arguments to a file inside the target directory so that when you generate the project in the sample place the same arguments are used.

[mr.bob](https://mrbob.readthedocs.org/en/latest/userguide.html) has this option:

```
  -w, --remember-answers
                        Remember answers to .mrbob.ini file inside output
                        directory
```

**Comment by jhermann** [comment]
Writing the file is already possible in a post hook:

https://github.com/Springerle/dh-virtualenv-mold/blob/master/hooks/post_gen_project.py#L58

The problem is that there is no easy way to read it in again (`--context-file` option – though adding that is easy).

**Comment by ionelmc** [comment]
Hmmmm, damnit that option is not implemented yet :grimacing: 

Still, I think cookiecutter should do this for you automatically.

**Comment by ionelmc** [comment]
Regarding https://github.com/audreyr/cookiecutter/pull/501 I had something else in mind. Because a have dozens of projects around made out of cookiecutter-pylibrary I update the older projects to have whatever fixes I do in cookiecutter-pylibrary (like new checks, fixes for setup.py, configuration improvements etc)  the `--replay` option don't help me at all :|

What if we'd have something like:
```
cookiecuttter gh:ionelmc/cookiecutter-library 
# a `old-project/.cookiecutterrc` would be saved automatically.

# and one year later  ...
cookiecuttter gh:ionelmc/cookiecutter-library --replay=old-project/.cookiecutterrc
```
@hackebrot what do you think?

**Comment by hackebrot** [comment]
I agree that ``--replay`` does not help you with your request of updating projects that have been created with ``cookiecutter``. Tbh, I am not really sure if there is a simple solution to that and also whether ``cookiecutter`` should be the tool to handle that.

You could have changed literally everything in your project after you created it, so it would be very hard to re-apply your manual changes.

MHhmhm... :neutral_face:

**Comment by hackebrot** [comment]
Can we close this guy? I think #501 meets the requirements of this very issue.

**Comment by ionelmc** [comment]
@hackebrot #501 - not really, the replay option don't take a configuration file as input (would be inconvenient anyway). Cookiecutter should automatically save the context in the destination directory and use that for the replay mode.

This bash script does what I want (only for `cookiecutter-pylibrary` projects):
```bash
#!/bin/bash -eEx

homedir=$(mktemp -d /tmp/ccutter-XXXXXX)
trap 'rm -rf "$homedir"' EXIT
projectdir="$1"
shift
cp $projectdir/.cookiecutterrc $homedir
HOME=$homedir cookiecutter "$@"
```
Eg: `cookiecutter.sh project-dir ionelmc/cookiecutter-pylibrary -f`

You can close it but I think the general public still need a feature that does what that shell script does.

**Comment by ionelmc** [comment]
Some update here, meanwhile I've made this: https://pypi.python.org/pypi/cookiepatcher

It works great but it's ugly and messed up on the inside (iow: it often breaks after a new cookiecutter release). It's a pain in the ass to maintain. Can we please have this functionality built into cookiecutter? 

My suggestion is to have a new option `--regenerate=project-path` and keep the same convention (a `.cookiecutterrc` yaml file that has the context). What do you think @hackebrot?

**Comment by ionelmc** [comment]
Another idea: retrofit replay feature to allow `--replay=project-path`.

**Comment by hackalog** [comment]
I can't give this enough thumbs up. 

Being able to specify a replay file would certainly solve a huge part of the problem, and I agree with @ionelmc : specifying a parameter to replay (an explicit path to the replay file) is probably the most obvious and backwards-compatible way to do it. 

i.e. if no option is specified (`--replay`) , implement the current behavior (look in the `replay_dir`). Otherwise, use the replay path specified (`--replay=path/to/replay-file.json`)

**Comment by ionelmc** [comment]
fyi I'm using https://pypi.org/project/cookiepatcher/ to solve this problem. It's an ugly solution that is coupled with cookiecutter internals but it works.

**Comment by Eronarn** [comment]
This is something I'd find very useful to be a part of core. Having a cookiecutterrc works fine for variables that are specific to the user or the machine, but fails for variables that must be set per-project.

**Comment by insspb** [comment]
Hi @ionelmc @Eronarn @hackalog @khorn @jhermann . 

I just merged pull request #906 that introduce `--replay-file` cli option. This option will be available in cookiecutter 2.0 this allow fast recreation from replay file. All you need to copy this file somewhere manually first. you even can distribute such file with you template. 

So I think this is solved. Closing.

**Comment by jhermann** [comment]
BTW, you can also do this: https://github.com/Springerle/py-generic-project/blob/master/hooks/post_gen_project.py#L179
